File: src/priv/bajdcc/LALR1/grammar/runtime/ObjectTools.java
package priv.bajdcc.LALR1.grammar.runtime;

import java.io.IOException;
import java.io.InputStream;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.io.OutputStream;

public class ObjectTools {
	@SuppressWarnings("unchecked")
	public static <T extends Object> T deserialize(InputStream input) {
		try {
			ObjectInputStream in = new ObjectInputStream(input);
			return (T) in.readObject();
		} catch (IOException | ClassNotFoundException e) {
			e.printStackTrace();
		}
		return null;
	}

	public static void serialize(Object object, OutputStream output) {
		try {
			ObjectOutputStream out = new ObjectOutputStream(output);
			out.writeObject(object);
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}

File: src/main/java/com/bajdcc/LALR1/grammar/runtime/ObjectTools.kt
package com.bajdcc.LALR1.grammar.runtime

import java.io.*

object ObjectTools {
    @Suppress("UNCHECKED_CAST")
    fun <T> deserialize(input: InputStream): T? {
        try {
            val objectInputStream = ObjectInputStream(input)
            return objectInputStream.readObject()!! as T
        } catch (e: IOException) {
            e.printStackTrace()
        } catch (e: ClassNotFoundException) {
            e.printStackTrace()
        }

        return null
    }

    fun serialize(any: Any, output: OutputStream) {
        try {
            val out = ObjectOutputStream(output)
            out.writeObject(any)
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }
}
